.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,20 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [master]
```

Comment:
Does this mean it only runs on `master` or also PRs to `master`?

I haven't used GitHub workflows.

Reply:
I assume that is correct. see: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#example-using-a-list-of-events.

Reply:
Hm, ok.  It'd be nice to have it work for PRs.  Open to giving this a whirl.